### PR TITLE
Second bunch of arm patches (v4)

### PR DESCRIPTION
**Pull request by gohai**
Rebased against master, supersedes #3555.

### CODE REVIEW

File: core/src/processing/opengl/PSurfaceJOGL.java [hunk 1]
```diff
@@ -195,7 +195,8 @@ protected void initGL() {
     if (profile == null) {
       if (PJOGL.profile == 2) {
         try {
-          profile = GLProfile.getGL2ES1();
+          // request at least GL2 or GLES2
+          profile = GLProfile.getGL2ES2();
```
Comment (benfry) [inline core/src/processing/opengl/PSurfaceJOGL.java]: @codeanticode If you can confirm we're safe to do this, I can merge this patch.

Comment (gohai) [inline core/src/processing/opengl/PSurfaceJOGL.java]: So much pressure.. :)

Comment (codeanticode) [inline core/src/processing/opengl/PSurfaceJOGL.java]: bumping the profile to GL2ES2 requires upgrading the shader code to [v330](https://github.com/mattdesl/lwjgl-basics/wiki/GLSL-Versions#version-330), at least on Mac OSX.

Comment (benfry) [inline core/src/processing/opengl/PSurfaceJOGL.java]: So we'll need to hold on this until https://github.com/processing/processing/issues/3505

Comment (benfry) [inline core/src/processing/opengl/PSurfaceJOGL.java]: But it sounds like that might take a little while, so let's add a platform check there and get the ES2 profile if it's running on ARM, and we can use that for the time being until #3505 is finished.

Comment (gohai) [inline core/src/processing/opengl/PSurfaceJOGL.java]: Great… will update and send a new PR.

**Comment by gohai** [comment]
@benfry Send new PR #3583

**Pull request closed without merging**